soccer-records-model/src/main/java/cz/muni/fi/pa165/team/TeamMatchService.java

```diff
@@ -0,0 +1,195 @@
+                        match.getId(),
+                        tm.getId(),
+                        match.getHomeTeam().getId(),
+                        startTime
+                    );
+                }
+                if (match.getAwayTeam().equals(tm.getAwayTeam()) ||
+                    match.getAwayTeam().equals(tm.getHomeTeam())) {
+                    throw new MatchTimeCollisionException(
+                        match.getId(),
+                        tm.getId(),
+                        match.getAwayTeam().getId(),
+                        startTime
+                    );
+                }
+            }
+        }
+
+        match.changeMatchTime(startTime, endTime);
+    }
+
+    /**
+     * This method sets new end time to the match.
+     *
+     * @param match match which end time should be updated
+     * @param endTime new end time of the match
+     */
+    public void endMatch(TeamMatch match, Date endTime)
+    {
+        Assert.notNull(match, "Cannot end the null match");
+        Assert.notNull(endTime, "Cannot end the match with a null end time");
+        Assert.isNull(match.getEndTime(), "Cannot end already ended match");
+
+        validateMatchTimes(match.getStartTime(), endTime);
+
+        match.endMatch(endTime);
+    }
+
+    /**
+     * This method creates new goal if all needed parameters are valid.
+     *
+     * @param goalsInActualMatch goals scored in the match
+     * @param scorer scorer of the goal
+     * @param assistant assistant of the goal
+     * @param match match in which the goal is scored
+     * @param matchTime time of the match in which the goal is scored
+     * @return newly created goal
+     */
+    public TeamMatchGoal addNewScoredGoal(
+        List<TeamMatchGoal> goalsInActualMatch,
+        TeamPlayer scorer,
+        TeamPlayer assistant,
+        TeamMatch match,
+        Date matchTime
+    )
+    {
+        Assert.notNull(scorer, "Cannot create new goal with a null scorer");
+        Assert.notNull(assistant, "Cannot create new goal with a null assistant");
+        Assert.isTrue(!scorer.equals(assistant), "Cannot create new goal with scorer and assistant who are the same player");
+
+        validateGoalMatchTime(matchTime, match);
+
+        if (!goalsInActualMatch.isEmpty()) {
+            for (TeamMatchGoal tmg : goalsInActualMatch) {
+                if (tmg.getScorer().equals(scorer) && tmg.getAssistant().equals(assistant)
+                    && tmg.getMatchTime().equals(matchTime)) {
+                    throw new GoalWithSameParametersAlreadyExistsException(
+                        scorer.getId(),
+                        assistant.getId(),
+                        match.getId(),
+                        matchTime
+                    );
+                }
+            }
+        }
+
+        return new TeamMatchGoal(scorer, assistant, match, matchTime);
+    }
+
+    private void validateMatchTimes(Date startTime, Date endTime)
```

Comment:
Bude to potřeba mít v konstruktoru te entity a stejně tak v těch metodach, ktery nastavuji ty jine časy... tzn asi vyčlenit do nějake private?